src/pool/srv_pool.c

```diff
@@ -4759,6 +4773,15 @@ evict_iter_cb(daos_handle_t ih, d_iov_t *key, d_iov_t *val, void *varg)
 		return -DER_IO;
 	}
 
+	/* If we specified a machine name as a filter check before we do the realloc */
+	if (arg->eia_machine) {
+		struct pool_hdl	*hdl = (struct pool_hdl *)val->iov_buf;
+
+		if (strncmp(arg->eia_machine, hdl->ph_machine, MAXHOSTNAMELEN) != 0) {
```

Comment:
`hdl->ph_machine` is an array, not a ptr, right? You could use `sizeof(hdl->ph_machine)`

Reply:
In this case I think that should be ok. It will effectively be the same thing since MAXHOSTNAMELEN+1 should always be \0.